subprojects/diagnostics/src/test/groovy/org/gradle/api/tasks/diagnostics/PropertyReportTaskTest.groovy

```diff
@@ -57,4 +57,36 @@ class PropertyReportTaskTest extends Specification {
         1 * renderer.addProperty("prop", "value")
         1 * renderer.addProperty("properties", "{...}")
     }
+
+    def "when a specific property is requested, only that property is requested"() {
+        when:
+        task.setProperty("prop")
+        task.generate(project)
+
+        then:
+        1 * project.properties >> ["prop": "value", "properties": "prop"]
+        1 * renderer.addProperty("prop", "value")
+        0 * renderer.addProperty("properties", "{...}")
+    }
+
+    def doesNotShowContentsOfThePropertiesPropertyWhenRequested() {
```

Comment:
The test method names should have the same formatting (normal spacing instead of camelCase)